### PR TITLE
DD Core Algorithms Cleanup

### PR DESCRIPTION
**Pull request by ianna**
* Replace CLHEP unit conversion with constexpr user-defined literals
* Update comments
* Merge plugin declaration and definition to reduce the number of files
* Remove obsolete file
* Use C++11 type aliases
* Use CMS coding style

Note: this PR is based on fully signed #24153

### CODE REVIEW

**Comment by ianna** [comment]
please test

File: DetectorDescription/Core/interface/DDUnits.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#ifndef DETECTOR_DESCRIPTION_DD_UNITS_H
+#define DETECTOR_DESCRIPTION_DD_UNITS_H
+
+#include <cmath>
+
+#define ConvertTo(_x, _y) (_x)/(1.0_##_y)
+
+constexpr long double _pi(M_PI);
+
+constexpr long double operator"" _deg( long double deg )
```
Comment (Dr15Jones) [inline DetectorDescription/Core/interface/DDUnits.h]: I believe all the examples of user defined types I've see have had them defined within a namespace. Perhaps using `namespace dd` would be appropriate?

Comment (Dr15Jones) [reply DetectorDescription/Core/interface/DDUnits.h]: See
https://akrzemi1.wordpress.com/2012/10/29/user-defined-literals-part-iii/

File: DetectorDescription/Core/interface/DDUnits.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#ifndef DETECTOR_DESCRIPTION_DD_UNITS_H
+#define DETECTOR_DESCRIPTION_DD_UNITS_H
+
+#include <cmath>
+
+#define ConvertTo(_x, _y) (_x)/(1.0_##_y)
```
Comment (Dr15Jones) [inline DetectorDescription/Core/interface/DDUnits.h]: To match CMS naming conventions, this should probably be `CONVERT_TO`

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
@Dr15Jones - thanks!

**Comment by ianna** [comment]
+1

**Comment by fabiocos** [comment]
@ianna for the test of this PR we are essentially relying on unit tests, as the relvals are normaly using DB snapshots, right?

**Pull request merged**